src/mongo/delegates/Views.class.php

```diff
@@ -848,4 +877,15 @@ protected function getCollectionForViewSpec($viewSpecId)
         return $this->getConfigInstance()->getCollectionForView($this->storeName, $viewSpecId);
     }
 
+    
+    /**
+     * Regenerate a specific composite document from its spec and root CBD
+     * @param $specification
+     * @param $compositeCollection
+     * @param $rootCbd
+     * @return void
+     */
+    public function regenerateOne($specification, $compositeCollection, $rootCbd) {
+        $this->saveGeneratedView($specification, $compositeCollection, $rootCbd, $specification['from'], $this->defaultContext);
```

Comment:
Because `generateView` doesn't only save; sometimes it queues for later.  When it does queue, it regenerates all composites for the CBD, not just the one we wish to target.  I will be using a queued approach in this PR eventually, but it should be targeted in what it regenerates.